Source/Scene/Model.js

```diff
@@ -4783,6 +4892,26 @@ function modifyShaderForClippingPlanes(
   return shader;
 }
 
+function modifyShaderForClippingPolygon(shader, clippingPolygon, context) {
+  shader = ShaderSource.replaceMain(shader, "gltf_clip_polygon_main");
+  var clippingFunc = getPolygonClippingFunction(clippingPolygon.union);
+  // TODO: We should avoid clip space entirely to avoid precision issues
+  //       with the clipspace -> ec  -> eyespace conversion. See Globe.VS
+  //       and how it forwards EC coordinates to the fragment shader if
+  //       clipping polygons for the globe are enabled.
+
+  var newMain =
+    "void main() {\n" +
+    "vec4 positionEC = czm_windowToEyeCoordinates(gl_FragCoord);\n" +
+    "positionEC /= positionEC.w;\n" +
```

Comment:
Completely by chance, I found that removing this line removes the fuzziness that @lilleyse and myself were seeing when using logarithmic depth buffers with tilesets at certain distances.
I'm not entirely sure why, but perhaps it's related to the things that get done to the Z in the vertex shader before log depth is written here in the fragment shader.

Reply:
Huh, and no side effects removing this line? Does it still work if log depth is disabled?

Reply:
So there weren't problems with the fuzziness issue with and without log depth, but it did break the Sandcastle example in this PR... the mysteries deepen. I'll keep investigating.

Reply:
Apologies for butting in, but could you briefly explain the reason for dividing the position by `w` in the first place? I'm not new to shaders, but this isn't clicking.

Reply:
@midnight-dev I'm not actually sure, it looks like other places where we use this particular overload of `czm_windowToEyeCoordinates` (like for the earlier [clipping planes shader code](https://github.com/CesiumGS/cesium/blob/master/Source/Scene/getClippingFunction.js#L44)) we don't do this divide-by-w. the math in that code precedes my involvement in Cesium's clipping planes, although the history is a little murky. There's a post on [Stackoverflow](https://stackoverflow.com/questions/13711252/what-does-gl-fragcoord-z-gl-fragcoord-w-represent) explaining that in `gl_FragCoord` the w can help you linearize depth, and is computed from `w` on `gl_Position` in the vertex shader, so I suspect `positionEC.w` serves a similar purpose. It's just we're already in the fragment shader, so there's maybe no need to actually divide by it? 

> but it did break the Sandcastle example in this PR... the mysteries deepen

@lilleyse I'm seeing for some reason that the proposed change here no longer breaks the shipping Sandcastle, which was kind of alarming.
I think I can chalk it up to some kind of browser cache thing, since I'm seeing that everything works fine in Firefox now as well.
Combined with the detail that we don't actually do this divide-by-w elsewhere in clipping code, I think it's safe to go ahead and apply this change, so imma do that.